### DISCUSSION TITLE
CTF

### DISCUSSION THREAD

**Discussion by hard1k7724** [Cyfrin/foundry-full-course-cu#2392] [Q&A] [answered]
Hey guys, I was writing the test suite for the first damn-vuln-defi CTF(Unstoppable) and I am getting this error - " Ran 1 test for test/unstoppable/Unstoppable.t.sol:UnstoppableChallenge
[FAIL. Reason: vm.startPrank: cannot overwrite a prank until it is applied at least once] test_unstoppable() "
Here is my test suite:
```
function test_unstoppable() public checkSolvedByPlayer {
        vm.startPrank(player);
        token.transfer(address(vault), 5e18);
        vm.stopPrank();

        vm.startPrank(monitor);
        vm.expectRevert(UnstoppableVault.InvalidBalance.selector);
        vault.flashLoan(
            monitorContract,
            address(token),
            INITIAL_PLAYER_TOKEN_BALANCE,
            ""
        );
        vm.stopPrank();
    }
    ```
    Can anyone tell me what am I doing wrong or how do fix this?

**Comment by EngrPips** [comment, marked as answer]
Do you have a prank inside of the `checkSolvedByPlayer` modifier?

**Reply by hard1k7724** [reply]
```
modifier checkSolvedByPlayer() {
        vm.startPrank(player, player);
        _;
        vm.stopPrank();
        _isSolved();
    }
```
Yes. Does that mean that I cant use the vm.prank in my test ?

**Reply by EngrPips** [reply]
Now it is glaring why you are getting that error. You didn't use the `vm.startPrank()` in the modifier, then you opened another one in the main body of the test function, and that is why the test reverted with `cannot overwrite a prank until it is applied at least once] test_unstoppable() ". ` What are you using the `vm.startPrank()` for in the modifier?

**Reply by hard1k7724** [reply]
It was already in the test file, I just didnt notice it, also I thought there is no point of editing anything in the file I guess. I just start started writing the solution.

**Reply by hard1k7724** [reply]
Also thanks, I edited the code and commented out the the
```
vm.startPrank(player)
```
The test is passing now.

**Reply by EngrPips** [reply]
Awesome, Keep crushing it.
